bpf/process/generic_calls.h

```diff
@@ -935,10 +935,15 @@ do_action(void *ctx, __u32 i, struct selector_action *actions, bool *post, bool
 		}
 		break;
 	case ACTION_OVERRIDE:
-		error = actions->act[++i];
 		if (enforce_mode) {
+#if defined(GENERIC_UPROBE) && defined(__TARGET_ARCH_x86)
+			index = actions->act[++i];
+			do_uprobe_override(ctx, index);
+#else
+			error = actions->act[++i];
 			do_override_action(error);
 			polacct = POLICY_OVERRIDE;
+#endif
```

Comment:
I don't think this is correct. For one, it does not  set `pollact = POLICY_OVERRIDE` for uprobes. 

Moreover, it does not update `i` in monitor mode. At the end of the function we do:
```C
	if (!err) {
		e->action = action;
		return ++i;
	}
```

So we need to maintain the correct value of `i` regardless of whether we are in monitor or enforce mode. That is, we need to do the `i++` in `error = actions->act[++i];` in both enforce and monitor cases.

So I think we need:
```C
	case ACTION_OVERRIDE:
		arg = actions->act[++i];
		if (enforce_mode) {
#if defined(GENERIC_UPROBE) && defined(__TARGET_ARCH_x86)
			do_uprobe_override(ctx, arg);
#else
			do_override_action(arg);
#endif
			polacct = POLICY_OVERRIDE;
		} else {
			polacct = POLICY_MONITOR_OVERRIDE;
		}
```

Reply:
@kkourt  ah right, how about:
```
                if (enforce_mode) {
#if defined(GENERIC_UPROBE) && defined(__TARGET_ARCH_x86)
                        index = actions->act[++i];
                        do_uprobe_override(ctx, index);
#else
                        error = actions->act[++i];
                        do_override_action(error);
#endif
                        polacct = POLICY_OVERRIDE;
                } else {
                        polacct = POLICY_MONITOR_OVERRIDE;
                }

```